scripts/lib/api/htmlToMd.ts

```diff
@@ -224,6 +180,73 @@ function removeEmphasisSpaces(
   }
 }
 
+function findNodeWithProperty(nodeList: any[], propertyName: string) {
+  return nodeList.find(
+    (child: any) => child.properties.className?.includes(propertyName),
+  );
+}
+
+function buildDiv(
+  childrenList: any[],
+  nodeClasses: string[],
+  handlers: Record<string, Handle>,
+  divType: "admonition" | "deprecated",
```

Comment:
Type defined to avoid calling the function with different divTypes. The logic of the function is the same but using a new helper function named findNodeWithProperty and merging the titleNode, title, and otherChildren definitions outside the ifs.

Reply:
After looking at this function for a while, I don't think I find it more readable because there is too much divergence between the two types:

* the `titleNode` search
* computing the `type`
* whether to call `toText` on the `title`, but I think that's actually unnecessary on line 211?

I think you could make this helper function worth it, though. Some suggestions:

* compute the `type` outside the helper function and pass it in
* explicitly pass in the title node to look for
* see if you can get rid of the `toText` in the return statement so that you can have one single `return`
* I'm trying to think of a more clear name than `buildDiv`, which isn't very precise. The best I'm thinking of is to rename `buildAdmonition` to be something more specific like `buildAdmonitionBase` and then change this to be `buildAdmonitionFromDiv`?